src/Events/Admin/Onboarding/Controller.php

```diff
@@ -88,6 +88,53 @@ public function handle_onboarding_page_dismiss(): void {
 		$this->container->make( Landing_Page::class )->handle_onboarding_page_dismiss();
 	}
 
+	/**
+	 * Redirects users to the Guided Setup page when accessing any TEC settings or management page for the first time.
+	 *
+	 * @since TBD
+	 *
+	 * @return void
+	 */
+	public function redirect_tec_pages_to_guided_setup(): void {
+		// Get some information about the page that we are trying to access.
+		$page      = tribe_get_request_var( 'page' );
+		$post_type = tribe_get_request_var( 'post_type' );
```

Comment:
```suggestion
		$page      = tec_get_request_var( 'page' );
		$post_type = tec_get_request_var( 'post_type' );
```
I would use the tec_ prefix where available, so we move away from tribe_ where we can.

Reply:
Also @sdokus a tip i would like to see amended!

Consider which condition bails faster!

e.g.
You are getting the page param from the URL.
You are getting the post_type param from the URL
you are quering the database for old TEC versions.
If there were older versions of TEC you bail.
!! If post_type does not contain tribe you bail !!
!! If page is equal to first-time-setup you bail !!

Is that the correct order of operations if you should the bail faster approach ?

for me it would be:

You are getting the page param from the URL.
If page is equal to first-time-setup you bail

You are getting the post_type param from the URL
If post_type does not contain tribe you bail

you are quering the database for old TEC versions.
If there were older versions of TEC you bail.